pkg/util/querylimits/limiter.go

```diff
@@ -0,0 +1,69 @@
+package querylimits
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+
+	"github.com/grafana/loki/pkg/util/limiter"
+	logutil "github.com/grafana/loki/pkg/util/log"
+)
+
+type Limiter struct {
+	logger log.Logger
+	limiter.CombinedLimits
+}
+
+func NewLimiter(log log.Logger, original limiter.CombinedLimits) *Limiter {
+	return &Limiter{
+		logger:         log,
+		CombinedLimits: original,
+	}
+}
+
+// MaxQueryLength returns the limit of the length (in time) of a query.
+func (l *Limiter) MaxQueryLength(ctx context.Context, userID string) time.Duration {
```

Comment:
Let's use `tenantID`. I know we already have a mixup of `user`, `userID`, `tenant`, `tenantID` and `orgID`. We should agree on one term, though ...

Reply:
Agreed. I've been meaning to resurrect the PR that changes all uses of these terms to `tenantID`.